engine-dmn/feel-juel/src/main/java/org/operaton/bpm/dmn/feel/impl/juel/transform/ListTransformer.java

```diff
@@ -17,52 +16,122 @@
 package org.operaton.bpm.dmn.feel.impl.juel.transform;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 
 import org.operaton.bpm.dmn.feel.impl.juel.FeelEngineLogger;
 import org.operaton.bpm.dmn.feel.impl.juel.FeelLogger;
 
+/**
+ * The ListTransformer class is responsible for transforming FEEL (Friendly Enough Expression Language)
+ * list expressions into JUEL (Java Unified Expression Language) expressions.
+ */
 public class ListTransformer implements FeelToJuelTransformer {
 
+  /** Logger instance for logging errors and debug information. */
   public static final FeelEngineLogger LOG = FeelLogger.ENGINE_LOGGER;
-  // regex to split by comma which does a positive look ahead to ignore commas enclosed in quotes
-  public static final String COMMA_SEPARATOR_REGEX = ",(?=([^\"]*\"[^\"]*\")*[^\"]*$)";
 
+  /**
+   * Determines if the given FEEL expression can be transformed.
+   *
+   * @param feelExpression the FEEL expression to check
+   * @return true if the expression can be transformed, false otherwise
+   */
   @Override
   public boolean canTransform(String feelExpression) {
     return splitExpression(feelExpression).size() > 1;
   }
 
+  /**
+   * Transforms a FEEL expression into a JUEL expression.
+   *
+   * @param transform the transformation logic to apply
+   * @param feelExpression the FEEL expression to transform
+   * @param inputName the name of the input variable
+   * @return the transformed JUEL expression
+   */
   @Override
   public String transform(FeelToJuelTransform transform, String feelExpression, String inputName) {
     List<String> juelExpressions = transformExpressions(transform, feelExpression, inputName);
     return joinExpressions(juelExpressions);
   }
 
+  /**
+   * Collects individual expressions from a FEEL list expression.
+   *
+   * @param feelExpression the FEEL expression to split
+   * @return a list of individual expressions
+   */
   protected List<String> collectExpressions(String feelExpression) {
     return splitExpression(feelExpression);
   }
 
+  /**
+   * Splits a FEEL expression into a list of sub-expressions based on the comma separator.
+   * Handles quoted strings to ensure commas within quotes are not treated as separators.
+   * <p>
+   * This method iterates through the characters of the FEEL expression.
+   * That way, it can prevent from StackOverflowError
+   * </p>
+   * @param feelExpression the FEEL expression to split
+   * @return a list of sub-expressions
+   */
   private List<String> splitExpression(String feelExpression) {
```

Comment:
This implementation is way more complex to understand than the previous one.

I would like to have dedicated unit tests for this method. To do so, make the method package protected, and maybe static. I would expect to have a parametrized test in ListTransformerTest takes various expressions and their splitted result as values.